### PR TITLE
Use dotnet test

### PR DESCRIPTION
**Pull request by jaredpar**
This change is migrating us away from the xUnit console runner and onto
the `dotnet test` command instead. This has the following advantages:

1. xunit console runner is not a supported product hence we really need to move off it anyways. 
2. Can take advantage of all the crash dump and blame analysis that they are doing in the dotnet test tool
3. Simplifies our story a bit because it means we are dealing with a single way to invoke tests across all the different configurations, OS we run. It's dotnet test all the time. This is true even when we are running .NET Framework tests. 
4. One step closer to removing all of the restore logic from our test jobs

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -266,5 +266,7 @@
       rather explicitly override it.
     -->
     <UsingToolMicrosoftNetCompilers Condition="'$(BootstrapBuildPath)' == ''">true</UsingToolMicrosoftNetCompilers>
+
+    <UseVSTestRunner>true</UseVSTestRunner>
```
Comment (RikkiGibson) [inline eng/Versions.props:270]: do we also need to update `MicrosoftNETTestSdkVersion` in this file?

Comment (jaredpar) [reply eng/Versions.props:270]: What relationship does that have to how the tests run here?

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL

File: src/Tools/Source/RunTests/ProcessTestExecutor.cs [hunk 2]
```diff
@@ -25,52 +26,68 @@ internal ProcessTestExecutor(TestExecutionOptions options)
...
+                        MaybeAddSeparator('&');
+                        builder.Append($"Trait!~{trait}");
+                    }
+                }
 
-            if (!string.IsNullOrWhiteSpace(Options.Trait))
-            {
-                var traits = Options.Trait.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
-                foreach (var trait in traits)
+                void MaybeAddSeparator(char separator = '|')
```
Comment (RikkiGibson) [inline src/Tools/Source/RunTests/ProcessTestExecutor.cs:65]: Please use the camelCase local function naming convention.

File: src/Workspaces/Core/Portable/FindSymbols/ReferenceLocationExtensions.cs [hunk 3]
```diff
@@ -39,6 +39,7 @@ internal static class ReferenceLocationExtensions
                         AddSymbols(semanticModel, documentGroup, result);
                     }
 
+                    // Keep compilation alive so that GetSemanticModelAsync remains cheap
```
Comment (RikkiGibson) [inline src/Workspaces/Core/Portable/FindSymbols/ReferenceLocationExtensions.cs:42]: Was this intended to be included by this PR?

File: src/Tools/Source/RunTests/ProcessTestExecutor.cs [hunk 4]
```diff
@@ -25,52 +26,68 @@ internal ProcessTestExecutor(TestExecutionOptions options)
...
             var assemblyName = Path.GetFileName(assemblyInfo.AssemblyPath);
-            var resultsFilePath = GetResultsFilePath(assemblyInfo);
-            var xmlResultsFilePath = Path.ChangeExtension(resultsFilePath, ".xml");
-            var htmlResultsFilePath = Path.ChangeExtension(resultsFilePath, ".html");
 
             var builder = new StringBuilder();
-            builder.AppendFormat(@"""{0}""", assemblyInfo.AssemblyPath);
-            builder.AppendFormat(@" {0}", assemblyInfo.ExtraArguments);
-            builder.AppendFormat($@" -xml ""{xmlResultsFilePath}""");
+            builder.Append($@"test");
```
Comment (RikkiGibson) [inline src/Tools/Source/RunTests/ProcessTestExecutor.cs:34]: might be worth commenting that this is about coming up with arguments to `dotnet` CLI in order to run this partition's tests, I had to blink at this for a sec to remember why we were passing the bare string `"test"` as a command line argument.

File: eng/Versions.props (hunk 1 above)
Comment (RikkiGibson) [reply eng/Versions.props:271]: I had the impression that `Microsoft.Net.Test.Sdk` actually controls what version of the test host gets loaded and that updating it is required to e.g. benefit from the change that makes the test host actually multi-target (or whatever was blocking us from moving to `dotnet test` before).

**Review by RikkiGibson — APPROVED** [review]
LGTM with some comments and nits

File: src/Tools/Source/RunTests/ProcessTestExecutor.cs (hunk 2 above)
Comment (sharwell) [reply src/Tools/Source/RunTests/ProcessTestExecutor.cs:65]: I believe the naming style for this directory is configured for PascalCase

Comment (jaredpar) [reply src/Tools/Source/RunTests/ProcessTestExecutor.cs:65]: This directory is PascalCased yes. 

We need to flip the repo to be consistent here now that .NET runtime came out with their recomendations so we can avoid this in the future.

File: src/Workspaces/Core/Portable/FindSymbols/ReferenceLocationExtensions.cs (hunk 3 above)
Comment (jaredpar) [reply src/Workspaces/Core/Portable/FindSymbols/ReferenceLocationExtensions.cs:42]: Yes. Just a comment update based on an investigation I did while looking at some of the test runs.

**Comment by jaredpar** [comment]
@sharwell do we need to look at the integration test here? Seems like a known issue.

**Comment by sharwell** [comment]
@jaredpar Second run hit a variation of the bug I mostly worked around with 6b43c8820f321a23f5f8d66f97e2e6f78ac8b920. Third run hit #48548. First run hit both of these plus a lower-priority flaky test that doesn't fail the double-pass often.

**Comment by jaredpar** [comment]
@sharwell so are we good to merge or do we need to retry again?

**Comment by sharwell** [comment]
@jaredpar looks good to me

**Pull request merged**